pkg/util/oidc/oidc.go

```diff
@@ -0,0 +1,48 @@
+package oidc
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"golang.org/x/oauth2"
+)
+
+func SubjectFromClaims(lg *slog.Logger, token *oauth2.Token, claimName string) *string {
+	rawToken, ok := token.Extra("id_token").(string)
```

Comment:
Don't parse this yourself; use https://github.com/rancher/opni/blob/131a150882dcb208b2eed500cb1ba570e9678d87/pkg/auth/openid/tokens.go#L38-L41